File: contiguous-array/contiguous-array.cs
public class Solution {
    public int FindMaxLength(int[] nums) {
        Dictionary<int,int> dict = new Dictionary<int,int>();
        dict.Add(0,-1);
        int max =0;
        int count =0;
        
        for(int i=0;i<nums.Length;i++)
        {
            count = count + (nums[i]==1 ?1 :-1);
            if(dict.ContainsKey(count))
            {
                max = Math.Max(max,i-dict[count]);
            }
            else
            {
                dict.Add(count,i);
            }
        }
        return max;
        
    }
}

File: assignments/src/Contiguous_array.java
import java.util.HashMap;

public class Contiguous_array {
    // Time: O(n), Space: O(n)
    public int findMaxLength(int[] nums) {
        int count = 0, maxLen = 0;
        HashMap<Integer, Integer> map = new HashMap();
        // count, index
        map.put(0,-1);

        for (int i = 0; i<nums.length; i++) {
            count += nums[i] == 1 ? 1 : -1;

            if (map.containsKey(count)) {
                maxLen = Math.max(maxLen, i - map.get(count));
            }
            else {
                map.put(count, i);
            }
        }
        return maxLen;
    }
}
